### PR TITLE
feat(sql,pandas): support from/until operators (comparison of days) TCTC-906 TCTC-907

### PR DESCRIPTION
**Pull request by davinov**
Implement the same functionnality than #1194 #1195 #1202 for pandas and SQL backends

Checklist:
- [x] set up test cases
- [x] support fixed dates in SQL translator 
- [x] support fixed dates in pandas executor 
- [x] ensure we do not use time in comparisons
- [x] support relative dates in SQL translator 
- [x] support relative dates in pandas executor

### CODE REVIEW

**Comment by davinov** [comment]
Ready! We can test it on the playground with Pandas, I'll make a second PR to add Snowflake to the playground

**Comment by davinov** [comment]
Not ready yet actually, I missed the implementaion of 2 operators in relative dates

File: server/src/weaverbird/backends/sql_translator/steps/utils/query_transformation.py [hunk 1]
```diff
@@ -9,6 +9,7 @@
     Condition,
     ConditionComboAnd,
     ConditionComboOr,
+    DateBoundCondition,
```
Comment (alice-sevin) [inline server/src/weaverbird/backends/sql_translator/steps/utils/query_transformation.py:12]: Pourquoi DateBound plutôt que RelativeDate ?

File: server/tests/backends/fixtures/filter/dates_from_until_strip_hours.json [hunk 2]
```diff
@@ -0,0 +1,104 @@
+{
+  "description": "From/until operators should always include the whole selected day. Users don't select hours (and shouldn't care about them in most cases).",
```
Comment (alice-sevin) [inline server/tests/backends/fixtures/filter/dates_from_until_strip_hours.json:2]: :+1:

File: server/src/weaverbird/backends/sql_translator/steps/utils/query_transformation.py (hunk 1 above)
Comment (davinov) [reply server/src/weaverbird/backends/sql_translator/steps/utils/query_transformation.py:12]: This type correspond to the whole condition, so column + operator + value.
The value maybe either fixed or relative (so not always relative)

File: server/tests/backends/fixtures/filter/relative_dates.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "description": "Dates from/until can be specified relative to another date.",
-  "exclude": ["sql", "snowflake"],
+  "exclude": ["sql"],
```
Comment (alice-sevin) [inline server/tests/backends/fixtures/filter/relative_dates.json:3]: Hum... le titre de la PR dit que c'est appliqué à sql aussi mais il est toujours exclu pour le moment non ?

**Review by alice-sevin — APPROVED** [review]
LGTM, a part le sql qui est toujours dans les exclude ?

**Comment by davinov** [comment]
@alice-sevin actually, it's normal to have `exclude: ["sql"]`
`sql` is for testing against mysql, while `snowflake` is for tetsing against snowflake.
We may put some effort to become compatible with mysql and postgresql, but for now these engines are not run in tests.

File: server/src/weaverbird/backends/pandas_executor/steps/utils/dates.py [hunk 4]
```diff
@@ -17,7 +17,14 @@ def evaluate_relative_date(relative_date: RelativeDate) -> datetime:
     else:
         raise NotImplementedError
 
+    if relative_date.duration == 'quarter':
+        quantity = relative_date.quantity * 3
+        duration = 'months'
+    else:
+        quantity = relative_date.quantity
+        duration = relative_date.duration + 's'
+
```
Comment (alice-sevin) [inline server/src/weaverbird/backends/pandas_executor/steps/utils/dates.py:26]: Est-ce qu'on a besoin de faire la même chose pour les semestre aussi @davinov ?

Comment (davinov) [reply server/src/weaverbird/backends/pandas_executor/steps/utils/dates.py:26]: Je crois qu'on pas les semestres dans la spec des dates relatives, si ?

Comment (davinov) [reply server/src/weaverbird/backends/pandas_executor/steps/utils/dates.py:26]: Check: pas de semestres ici: https://github.com/ToucanToco/weaverbird/blob/master/server/src/weaverbird/pipeline/dates.py#L11 https://github.com/ToucanToco/weaverbird/blob/master/src/lib/dates.ts#L19

**Pull request merged**